File: Arrays/CPP/ContainsDuplicates.cpp
/*
Leetcode Question :
https://leetcode.com/problems/contains-duplicate
*/

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

bool containsDuplicate(vector<int>& nums) {
    sort(nums.begin(), nums.end());
    for (size_t i = 0; i < nums.size() - 1; ++i) {
        if (nums[i] == nums[i + 1]) {
            return true;
        }
    }
    return false;
}

int main() {
    vector<int> nums = {1};
    cout << boolalpha << containsDuplicate(nums) << endl;
    return 0;
}

File: Arrays/Java/ContainsDuplicates.java
/*
Leetcode Question :
https://leetcode.com/problems/contains-duplicate
*/
import java.util.Arrays;

public class ContainsDuplicates {
    public static boolean containsDuplicate(int[] nums) {
        Arrays.sort(nums);
        for(int i = 0; i<nums.length - 1; i++) {
            if(nums[i] == nums[i+1]) {
                return true;
            }
        }
        return false;
    }
    public static void main(String[] args) {
        int [] nums = {1};
        System.out.println(containsDuplicate(nums));
    }
}
